crates/polars-core/src/series/any_value.rs

```diff
@@ -493,9 +493,17 @@ fn any_values_to_list(
     let mut valid = true;
     #[allow(unused_mut)]
     let mut out: ListChunked = if inner_type == &DataType::Null {
+        let mut first_value_dtype = DataType::Null;
         avs.iter()
             .map(|av| match av {
-                AnyValue::List(b) => Some(b.clone()),
+                AnyValue::List(b) => {
+                    if !b.is_empty() && matches!(first_value_dtype, DataType::Null) {
```

Comment:
I am a little concerned about performance here. 

Can you do something like this? That makes the branch prediction easier.
```
let mut set_first = false;
let dtype = b.dtype();|

// set first value
if !set_first && first_value_dtype != dtype {
   first_value_dtype = dtype;
   set_first = true;
   // after this branch prediction will always be correct
}

if b.is_empty() || b.null_count() == b.len() {
   ...
}


```

Reply:
Done. 

I think concerns are justified. There's already some branching in `from_iter` and we're adding even more.